src/intervals/interval_operations/boolean.jl

```diff
@@ -4,200 +4,331 @@
 """
+isbounded(x::BareInterval) = (isfinite(inf(x)) & isfinite(sup(x))) | isempty_interval(x)
+
+function isbounded(x::Interval)
+    isnai(x) && return false
+    return isbounded(bareinterval(x))
+end
+
+"""
+    isbounded(x::BareInterval)
+    isbounded(x::Interval)
+
+Test whether `x` is not empty and has infinite bounds.
+"""
+isunbounded(x::BareInterval) = !isbounded(x)
+
+function isunbounded(x::Interval)
+    isnai(x) && return false
+    return isunbounded(bareinterval(x))
+end
+
+"""
+    isnai(x::BareInterval)
+    isnai(x::Interval)
+
+Test whether `x` is an NaI.
+"""
+isnai(::BareInterval) = false
+
+isnai(x::Interval) = decoration(x) == ill
+
+"""
+    iscommon(x::BareInterval)
+    iscommon(x::Interval)
+
+Test whether `x` is non-empty and bounded.
+"""
+iscommon(x::BareInterval) = !(isentire_interval(x) | isempty_interval(x) | isunbounded(x))
+
+function iscommon(x::Interval)
+    isnai(x) && return false
+    return iscommon(bareinterval(x))
+end
+
+"""
+    isatomic(x::BareInterval)
     isatomic(x::Interval)
 
-Check whether an interval `x` is *atomic*, i.e. is unable to be split.
-This occurs when the interval is empty, or when the upper bound equals the lower
-bound or the bounds are consecutive floating point numbers.
+Test whether `x` is unable to be split. This occurs when the interval is empty,
+or when the upper bound equals the lower bound or the bounds are consecutive
+floating-point numbers.
 """
-isatomic(x::Interval) = isempty_interval(x) || (inf(x) == sup(x)) || (sup(x) == nextfloat(inf(x)))
+isatomic(x::BareInterval) = isempty_interval(x) | (inf(x) == sup(x)) | (sup(x) == nextfloat(inf(x)))
+
+function isatomic(x::Interval)
+    isnai(x) && return false
+    return isatomic(bareinterval(x))
+end
 
 """
-    isthinzero(x)
+    isthin(x::BareInterval)
+    isthin(x::Interval)
+
+Test whether `x` contains only a real.
 
-Return whether the interval only contains zero.
+Implement the `isSingleton` function of the IEEE Standard 1788-2015 (Table 9.3).
 """
-isthinzero(x::Interval) = iszero(inf(x)) && iszero(sup(x))
+isthin(x::BareInterval) = inf(x) == sup(x)
+
+function isthin(x::Interval)
+    isnai(x) && return false
+    return isthin(bareinterval(x))
+end
 
 """
-    isthin(a::Interval, x::Number)
+    isthin(x::BareInterval, y::Number)
+    isthin(x::Interval, y::Number)
 
-Check if the interval `a` contains exactly (and only) the number `x`.
+Test whether `x` contains only `y`.
 """
-isthin(a::Interval, x::Real) = inf(a) == sup(a) == x
+isthin(x::BareInterval, y::Number) = inf(x) == sup(x) == y
+
+function isthin(x::Interval, y::Number)
+    isnai(x) && return false
+    return isthin(bareinterval(x), y)
+end
+
+"""
+    isthinzero(x::BareInterval)
+    isthinzero(x::Interval)
+
+Test whether `x` contains only zero.
+"""
+isthinzero(x::BareInterval) = iszero(inf(x)) & iszero(sup(x))
+
+function isthinzero(x::Interval)
+    isnai(x) && return false
+    return isthinzero(bareinterval(x))
+end
```

Comment:
This function remainded me about `iszero`, which seems to work for `BareInterval`s (!), though I don't see how:
```julia
julia> iszero(bareinterval(0.1))
false

julia> @which iszero(bareinterval(0.1))
iszero(x)
     @ Base number.jl:42

julia> iszero(interval(0.1))
ERROR: ArgumentError: `==` is purposely not supported for intervals. See instead `isequal_interval`.
Stacktrace:
 [1] ==(#unused#::Interval{Float64}, #unused#::Interval{Float64})
   @ IntervalArithmetic ~/.julia/dev/IntervalArithmetic/src/intervals/real_interface.jl:70
 [2] iszero(x::Interval{Float64})
   @ Base ./number.jl:42
 [3] top-level scope
   @ REPL[86]:1
```

This comment is therefore has two points:
- Either `iszero` shouldn't work with `BareInterval` (at least not if it involves promotion) or at all, and thus favor using `isthinzero`, or we should rely on it completely (and remove `isthinzero`).
- The error thrown by `iszero(::Interval)` shows the use of `==`, instead of `isequal_interval`, which should be corrected.